src/main/java/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -179,6 +179,10 @@ static MoveValidationResult validateFirst(final GameData data, final Collection<
     for (final Unit unit : units) {
       if (TripleAUnit.get(unit).getSubmerged()) {
         result.addDisallowedUnit("Cannot move submerged units", unit);
+        continue;
+      }
+      if (Matches.unitIsDisabled().test(unit)) {
+        result.addDisallowedUnit("Cannot move disabled units", unit);
```

Comment:
Would an `else if` obviate the need for the `continue` short-circuit?

```java
for (final Unit unit : units) {
  if (TripleAUnit.get(unit).getSubmerged()) {
    result.addDisallowedUnit("Cannot move submerged units", unit);
  } else if (Matches.unitIsDisabled().test(unit)) {
    result.addDisallowedUnit("Cannot move disabled units", unit);
  }
}
```

Reply:
Yeah, I'm an idiot. Added the if first and then realized that you wouldn't want to add the same unit twice. Updated it.